test/mocha/20-resources.js

```diff
@@ -231,11 +231,25 @@ describe('resources', function() {
     });
 
   it('should acquire with updated restriction', async function() {
+    const id = await generateId();
+    await restrictions.insert({
+      restriction: {
+        id,
+        zone: ZONES.ONE,
+        resource: RESOURCES.ASPARAGUS,
+        method: 'limitOverDuration',
+        methodOptions: {
+          limit: 1,
+          duration: 'P30D'
+        }
+      }
+    });
     // change restriction to allow additional acquisition
     await restrictions.update({
       restriction: {
+        id,
         zone: ZONES.ONE,
-        resource: RESOURCES.ORANGE,
+        resource: RESOURCES.ASPARAGUS,
```

Comment:
noting that it seems odd to me that a restriction might apply to oranges one day and asparagus the next.  Seems maybe the resource would be an immutable property of the restriction.  Not sure what the business logic is around how restrictions are used in practice.

Reply:
We are inserting the restriction here, it was updated to `ASPARAGUS` to avoid interfering with the limits of the other tests.

Reply:
Sure, I'm not taking issue with what is being done here in the test.  I'm questioning the fact that it is possible to change the `resource` for a restriction.  It seems to me that if one wants to have a restriction on a different resource, one creates a new restriction, not alter an existing one.  If @dlongley doesn't see any issue here, then there is no cause for concern.  Whatever the case, the issue is not with this test, it's with the restriction update API business rules.

Reply:
It's technically fine to make this sort of change (just suppose the restriction was made in error to begin with and the resource needs to be updated). The only potential problem is people seeing this as a the main use of `update` (which it probably isn't).

Reply:
Agreed on the logic. At this point, when updating a restriction, the entire restriction is passed in the body, and the updates are applied to the database. Previously the restriction was queried via `zone` and `resource` so it was not possible to update those properties, but since it is queried by `id` it now would be possible to change the `resource`. Is this something that should be dealt with in this PR, or should I open an issue detailing this for future work?

Reply:
I don't think we need to do anything with it, this is a low-level API and there are conceivable use cases where you'd need to change the resource, though they should be rare.